platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/history/changes/NonIncrementalInputChanges.java

```diff
@@ -57,7 +57,7 @@ public Iterable<FileChange> getFileChanges(Provider<? extends FileSystemLocation
 
     public Iterable<FileChange> getObjectFileChanges(Object parameter) {
         CurrentFileCollectionFingerprint currentFileCollectionFingerprint = currentInputs.get(incrementalInputProperties.getPropertyNameFor(parameter));
```

Comment:
💅 Maybe move the null check to the initialization, so the failure will be closer to the source?

```java
        CurrentFileCollectionFingerprint currentFileCollectionFingerprint = Objects.requireNonNull(currentInputs.get(incrementalInputProperties.getPropertyNameFor(parameter)));
        return () -> getAllFileChanges(currentFileCollectionFingerprint).iterator();
```